### PR TITLE
CT-1668 full text search test data

### PR DESCRIPTION
**Pull request by beccapearce**

### CODE REVIEW

File: db/structure.sql [hunk 1]
```diff
@@ -597,6 +597,36 @@ CREATE TABLE schema_migrations (
 );
 
 
+--
+-- Name: search_index; Type: TABLE; Schema: public; Owner: -
+--
+
+CREATE TABLE search_index (
```
Comment (misaka) [inline db/structure.sql]: Don't think this should be here.

File: db/structure.sql [hunk 2]
```diff
@@ -972,6 +1002,13 @@ ALTER TABLE ONLY report_types ALTER COLUMN id SET DEFAULT nextval('report_types_
 ALTER TABLE ONLY reports ALTER COLUMN id SET DEFAULT nextval('reports_id_seq'::regclass);
 
 
+--
+-- Name: id; Type: DEFAULT; Schema: public; Owner: -
+--
+
+ALTER TABLE ONLY search_index ALTER COLUMN id SET DEFAULT nextval('search_index_id_seq'::regclass);
```
Comment (misaka) [inline db/structure.sql]: This also shouldn't be here.

File: lib/tasks/search_seed_data.rake [hunk 3]
```diff
@@ -0,0 +1,16 @@
+namespace :seed do
+  namespace :search do
+    desc 'Seeds the database ready for testing full text search'
+    task data: :environment do
+      raise "Cannot run this task on the production environment" if HostEnv.prod?
+
+      # Rake::Task['db:reseed'].invoke
```
Comment (misaka) [inline lib/tasks/search_seed_data.rake]: Still need this?

**Review by misaka — CHANGES_REQUESTED** [review]

File: db/seeders/search_test_data_seeder.rb [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        puts "with these errors: "
+        puts @case.errors.full_messages
+      end
+    end
+  end
+
+  def select_type
+    types = [
+              "Case::FOI::Standard",
+              "Case::FOI::ComplianceReview",
```
Comment (stephenrichards) [inline db/seeders/search_test_data_seeder.rb]: Just a note on efficiency.  Every time select_type (and a number of other methods) is called, you are instantiating a number of strings, and then instantiating an array, and it's the same strings and same array every time.  Better to do 
```@names ||= [ 'name 1', name 2'....]```

**Review by stephenrichards — APPROVED** [review]

**Review by misaka — APPROVED** [review]

**Pull request merged**